service/topology_coordinator.cc

```diff
@@ -1052,9 +1052,12 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                         }
                     }
                     if (do_barrier()) {
-                        rtlogger.debug("Will set tablet {} stage to {}", gid, locator::tablet_transition_stage::write_both_read_old);
+                        auto next_transition_stage = (trinfo.transition == locator::tablet_transition_kind::rf_change)
```

Comment:
Why is that so? You skip the whole transition state machone and switch allow_write_both_read_old -> end_migration instantly???

Reply:
This is per https://github.com/scylladb/scylladb/pull/16723#discussion_r1456605027
CC @tgrabiec

Reply:
I think this place needs a comment explaining why the regular state change is disturbed here and where/when it really happens

Reply:
I'm still discussing with @tgrabiec how the transition should look like. From what I am guessing, it may be that a better approach would be to get to the `streaming` step and only then jump to the `end_migration`. Once that's clarified, I'll provide an explanatory comment.